services/tables/src/main/java/com/linkedin/openhouse/tables/audit/TableAuditAspect.java

```diff
@@ -195,6 +195,46 @@ protected ApiResponse<Void> auditDeleteTable(
     return result;
   }
 
+  /** Install the Around advice for deleteTable() method in OpenHouseTablesApiHandler */
+  @Around(
+      "execution("
+          + "com.linkedin.openhouse.common.api.spec.ApiResponse<Void> "
+          + "com.linkedin.openhouse.tables.api.handler.TablesApiHandler.renameTable(..)) "
+          + "&& args(fromDatabaseId, fromTableId, toDatabaseId, toTableId, actingPrincipal)")
+  protected ApiResponse<Void> auditRenameTable(
+      ProceedingJoinPoint point,
+      String fromDatabaseId,
+      String fromTableId,
+      String toDatabaseId,
+      String toTableId,
+      String actingPrincipal)
+      throws Throwable {
+    ApiResponse<Void> result = null;
+    TableAuditEvent fromEvent =
+        TableAuditEvent.builder()
+            .eventTimestamp(Instant.now())
+            .databaseName(fromDatabaseId)
+            .tableName(fromTableId)
+            .operationType(OperationType.RENAME_FROM)
+            .build();
+    TableAuditEvent toEvent =
+        TableAuditEvent.builder()
+            .eventTimestamp(Instant.now())
+            .databaseName(toDatabaseId)
+            .tableName(toTableId)
+            .operationType(OperationType.RENAME_TO)
+            .build();
+    try {
+      result = (ApiResponse<Void>) point.proceed();
```

Comment:
So after rename events about the previous version are disconnected from events on the new tableId?

Reply:
Yes from the consumer POV `RENAME_FROM` would act functionally the same as a DROP and `RENAME_TO` would act like a CREATE, hence adding 2 events